File: app/src/main/java/com/foxek/simpletimer/utils/IntervalUtils.java
package com.foxek.simpletimer.utils;

public class IntervalUtils {

    public static int convertToSeconds(String minute, String seconds) {
        return Integer.valueOf(minute) * 60 + Integer.valueOf(seconds);
    }

    public static String formatEditTextData(int time) {
        if (time < 10)
            return "0" + String.valueOf(time);
        else
            return String.valueOf(time);
    }

    public static String formatIntervalData(int time) {
        int minutes, seconds;
        minutes = time / 60;
        seconds = time % 60;

        if ((minutes < 10) && (seconds < 10)) {
            if ((minutes == 0) && (seconds < 0))
                return "00" + ":" + "00";
            else
                return "0" + String.valueOf(minutes) + ":" + "0" + String.valueOf(seconds);
        } else if ((minutes < 10) && (seconds >= 10))
            return "0" + String.valueOf(minutes) + ":" + String.valueOf(seconds);
        else if ((minutes >= 10) && (seconds < 10))
            return String.valueOf(minutes) + ":" + "0" + String.valueOf(seconds);
        else
            return String.valueOf(minutes) + ":" + String.valueOf(seconds);
    }

    public static String formatIntervalNumber(int number) {
        String rawString = String.valueOf(number);
        char lastNumber = rawString.charAt(rawString.length() - 1);
        switch (lastNumber) {
            case '2':
            case '6':
            case '7':
            case '8':
                return rawString + "-ой подход";
            case '3':
                return rawString + "-ий подход";
            default:
                return rawString + "-ый подход";
        }
    }
}

File: app/src/main/java/com/foxek/simpletimer/utils/IntervalUtils.kt
package com.foxek.simpletimer.utils

import java.util.*

fun formatIntervalData(time: Int): String {
    val minutes: Int = time / 60
    val seconds: Int = time % 60

    return if (minutes < 10 && seconds < 10) {
        if (minutes == 0 && seconds < 0)
            "00" + ":" + "00"
        else
            "0$minutes:0$seconds"
    } else if (minutes < 10 && seconds >= 10)
        "0$minutes:$seconds"
    else if (minutes >= 10 && seconds < 10)
        "$minutes:0$seconds"
    else
        "$minutes:$seconds"
}

fun formatIntervalNumber(number: Int): String {
    return if (Locale.getDefault().displayLanguage == "русский")
        cyrillicOrdinal(number)
    else
        latinOrdinal(number)
}

fun cyrillicOrdinal(number: Int): String {
    val rawString = number.toString()
    return when (rawString[rawString.length - 1]) {
        '2', '6', '7', '8' -> "$rawString-ой подход"
        '3' -> "$rawString-ий подход"
        else -> "$rawString-ый подход"
    }
}

fun latinOrdinal(number: Int): String {
    val sufixes = arrayOf("th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th")
    return when (number % 100) {
        11, 12, 13 -> "${number}th interval"
        else -> "$number${sufixes[number % 10]} interval"
    }
}

